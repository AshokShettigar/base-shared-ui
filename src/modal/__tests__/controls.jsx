import { shallow } from 'enzyme';
import { BUTTON_VARIATION } from 'platform-common-ui';
import Footer from 'platform-common-ui/modal/controls';

describe('modal/controls', function () {
    it('should render without controls', function () {
        const wrapper = shallow(<Footer />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render the controls', function () {
        const handler = () => true;
        const controls = [{
            label: 'Save',
            onClick: handler,
            variation: BUTTON_VARIATION.PRIMARY
        }, {
            label: 'Cancel',
            onClick: handler,
            variation: BUTTON_VARIATION.SECONDARY
        }];
        const wrapper = shallow(<Footer controls={controls} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render the controls with state loading', function () {
        const handler = () => true;
        const controls = [{
            label: 'Save',
            onClick: handler,
            variation: BUTTON_VARIATION.PRIMARY,
            isLoading: true
        }, {
            label: 'Cancel',
            onClick: handler,
            variation: BUTTON_VARIATION.SECONDARY
        }];
        const wrapper = shallow(<Footer controls={controls} />);

        expect(wrapper).toMatchSnapshot();
    });
});
