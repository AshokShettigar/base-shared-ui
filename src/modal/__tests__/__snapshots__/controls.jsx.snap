// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal/controls should render the controls 1`] = `
<div
  className="cn-modal-controls"
  data-component="ModalControls"
>
  <Button
    className="cn-modal-controls__button"
    isDark={false}
    isDisabled={false}
    onClick={[Function]}
    size={1}
    type="button"
    variation={1}
  >
    Save
  </Button>
  <Button
    className="cn-modal-controls__button"
    isDark={false}
    isDisabled={false}
    onClick={[Function]}
    size={1}
    type="button"
    variation={2}
  >
    Cancel
  </Button>
</div>
`;

exports[`modal/controls should render the controls with state loading 1`] = `
<div
  className="cn-modal-controls"
  data-component="ModalControls"
>
  <Button
    className="cn-modal-controls__button"
    isDark={false}
    isDisabled={false}
    isLoading={true}
    onClick={[Function]}
    size={1}
    type="button"
    variation={1}
  >
    Save
  </Button>
  <Button
    className="cn-modal-controls__button"
    isDark={false}
    isDisabled={false}
    onClick={[Function]}
    size={1}
    type="button"
    variation={2}
  >
    Cancel
  </Button>
</div>
`;

exports[`modal/controls should render without controls 1`] = `
<div
  className="cn-modal-controls"
  data-component="ModalControls"
/>
`;
